manager/constraint/constraint.go

```diff
@@ -125,6 +125,26 @@ func NodeMatches(constraints []Constraint, n *api.Node) bool {
 			if !constraint.Match(n.Spec.Role.String()) {
 				return false
 			}
+		case strings.EqualFold(constraint.key, "node.platform.os"):
+			if n.Description == nil || n.Description.Platform == nil {
+				if !constraint.Match("") {
+					return false
+				}
```

Comment:
I was thinking use cases like IoT devices where platform is not recognized by code and it's skipped. `node.platform.os != linux` would select them if they are mixed with Linux servers.